prog/postgres/postgres_nexus.rb

```diff
@@ -0,0 +1,184 @@
+# frozen_string_literal: true
+
+require "forwardable"
+
+require_relative "../../lib/util"
+
+class Prog::Postgres::PostgresNexus < Prog::Base
+  extend Forwardable
+  def_delegators :server, :vm
+
+  semaphore :restart, :destroy
+
+  def self.assemble(project_id, location, server_name, vm_size, storage_size_gib)
+    unless (project = Project[project_id])
+      fail "No existing project"
+    end
+
+    Validation.validate_vm_size(vm_size)
+    Validation.validate_name(server_name)
+    Validation.validate_location(location, project.provider)
+
+    DB.transaction do
+      ubid = PostgresServer.generate_ubid
+
+      ssh_key = SshKey.generate
+      vm_st = Prog::Vm::Nexus.assemble(
+        ssh_key.public_key,
+        Config.postgres_service_project_id,
+        location: location,
+        name: ubid.to_s,
+        size: vm_size,
+        storage_volumes: [
+          {encrypted: true, size_gib: 30},
+          {encrypted: true, size_gib: storage_size_gib}
+        ],
+        boot_image: "ubuntu-jammy",
+        enable_ip4: true
+      )
+
+      Sshable.create(
+        unix_user: "ubi",
+        host: "temp_#{vm_st.id}",
+        raw_private_key_1: ssh_key.keypair
+      ) { _1.id = vm_st.id }
+
+      server = PostgresServer.create(
+        project_id: project_id, location: location, server_name: server_name,
+        target_vm_size: vm_size, target_storage_size_gib: storage_size_gib,
+        superuser_password: SecureRandom.base64(15),
+        vm_id: vm_st.id
+      ) { _1.id = ubid.to_uuid }
+      server.associate_with_project(project)
+
+      Strand.create(prog: "Postgres::PostgresNexus", label: "start") { _1.id = server.id }
+    end
+  end
+
+  def before_run
+    when_destroy_set? do
+      if strand.label != "destroy"
+        server.active_billing_records.each do |br|
+          br.update(span: Sequel.pg_range(br.span.begin...Time.now))
+        end
+
+        hop_destroy
+      end
+    end
+  end
+
+  label def start
+    nap 5 unless vm.strand.label == "wait"
+    vm.sshable.update(host: vm.ephemeral_net4)
+    hop_bootstrap_rhizome
+  end
+
+  label def bootstrap_rhizome
+    register_deadline(:wait, 10 * 60)
+
+    bud Prog::BootstrapRhizome, {"target_folder" => "postgres", "subject_id" => vm.id, "user" => "ubi"}
+    hop_wait_bootstrap_rhizome
+  end
+
+  label def wait_bootstrap_rhizome
+    reap
+    hop_mount_data_disk if leaf?
+    donate
+  end
+
+  label def mount_data_disk
+    case vm.sshable.cmd("common/bin/daemonizer --check format_disk")
+    when "Succeeded"
+      vm.sshable.cmd("sudo mkdir -p /dat")
+      device_path = vm.vm_storage_volumes.find { _1.boot == false }.device_path.shellescape
+
+      vm.sshable.cmd("sudo mount #{device_path} /dat")
```

Comment:
Should we add `fstab` record for it? As I know mount not persistent, so it will not mount after a reboot.

Reply:
This was in my notes, but then I forgot about it. Thanks for catching it. I'll look into it.

Reply:
Done. I added a utility to safely adding records to fstab https://github.com/ubicloud/ubicloud/pull/578/commits/4904a4709e4ea9f05bd402db95759ef3a79de764